fastlane/lib/fastlane/helper/ci_helper.rb

```diff
@@ -0,0 +1,33 @@
+module Fastlane
+  module Helper
+    class CI
+      def self.should_run?(force: false)
```

Comment:
This doesn't belong in `Helper::CI` IMO, or at least not with `should_run?` as the name

Reply:
Hey @dantoml 👋 hmm, why is that? 🤔 Could you please suggest a new name for this method if in your opinion the current one is wrong? 🙏 Thanks!

Reply:
👋, So with `CI.should_run?` feels like inverting the thing we want to actually do. We're not asking if a `CI` thing should run, we have actions that should only run in CI or if forced.

Personally, I'd keep this specific function in the actions, to avoid accidental unintended consequences of changing this method.

Reply:
Keeping this specific function in the actions would be copy-paste all over the place - imagine making the change there one day. 

What do you think about adding `should_run_on_ci?(force: false)` to `actions_helper.rb` file:

```
# actions_helper.rb

module Fastlane
  module Actions
      def self.should_run_on_ci?(force: false)
        Helper.ci? || force
      end
  end
end
```
? 🤔

Reply:
The fastlane model doesn't really give us a nice way to add these types of filters generically, because ideally this would be some type of trait or action filter, but that's not the reality we have today.

Extracting a single `||` into a helper isn't really providing us any value IMO, and I don't think we can give it a name that is actually what we're asking.

Reply:
This helper is more than just `Helper.ci? || force`. It also has the `setup_keychain` method. But... I guess maybe this doesn't belong in helpers and should end up being a standalone action??? 🤔 

And then this `setup_ci_keychain` action is called by `setup_circle_ci` and `setup_travis_ci`? Thoughts?

Reply:
Sounds good to me 👍 Does this mean we are going to expose `setup_ci_keychain` to fastlane user (if so I guess having `ci_helper.rb` would prevent that)? 🤔 

Also let's keep in mind the future improvements pointed out in [this comment](https://github.com/fastlane/fastlane/pull/14346#issuecomment-470657627). We might end up with having more small CI related actions - is that okay for us? 🤔

Reply:
@mollyIV Maybe we just 💣 this helper and create the new aciton with the name of `setup_ci` instead? 🤔 That way it can flex into doing other things in the future if needed?

@dantoml Thoughts? 😇

Reply:
@joshdholtz `setup_ci` was our last step in this plan: https://github.com/fastlane/fastlane/pull/14346#issuecomment-470657627. 

However I guess you are right and why not introduce it right away? 🤔 

To briefly sum up in a code snippets that everyone'd one the same page:

```ruby
# setup_travis.rb file

def self.run(params)
    Fastlane::Actions.setup_ci(provider: "travis", force: params[:force])
end
```

```ruby
# setup_circle_ci.rb file

def self.run(params)
    Fastlane::Actions.setup_ci(provider: "circleci", force: params[:force])
end
```

```ruby
# setup_ci.rb file

def self.run(params)
    force = params[:force]
    provider = params[:provider]

    unless should_run?(force: force)

    case provider
    when 'travis'
      # Do travis specific stuff: setup_keychain
    when 'circleci'
      # Do circleci specific stuff: setup_keychain, setup_output_paths
    else
      # Shall we throw an exception here and fail the action? 
end
```

@joshdholtz @dantoml please share your thoughts 🙏

Reply:
If we really want to introduce another layer of abstraction to this (I'm still not sure this is worth DRYing), the most ruby like abstraction would be to have a `ci_setup` base class, that implements `setup_keychain` and `should_run`, with `setup_circle_ci` and `setup_travis` subclasses, that can then override `self.run` with

```ruby
def self.run(params)
  unless should_run?(force: params[:force])
    ...
  end

  setup_keychain(params)
  setup_output_paths
end
```

```ruby
def self.run(params)
  unless should_run?(force: params[:force])
    ...
  end

  setup_keychain(params)
end
```

Arguably the superclass run implementation could default to should_run... setup_keychain, but that would be overly abstract probably.

Reply:
I like what @mollyIV posted above - simple, clear and obvious. (Only exception: Take the `setup_keychain` out of the `case` if it is valid for all cases)